v2/tools/asoctl/internal/importing/arm_resource_importer.go

```diff
@@ -0,0 +1,242 @@
+	//req, err := runtime.NewRequest(ctx, http.MethodGet, runtime.JoinPaths(rmConfig.Endpoint, urlPath))
+	req, err := azruntime.NewRequest(ctx, http.MethodGet, azruntime.JoinPaths(ri.serviceConfiguration.Endpoint, armID))
+	if err != nil {
+		return nil, err
+	}
+
+	requestQueryPart := req.Raw().URL.Query()
+	requestQueryPart.Set("api-version", apiVersion)
+	req.Raw().URL.RawQuery = requestQueryPart.Encode()
+
+	req.Raw().Header.Set("Accept", "application/json")
+
+	klog.V(3).Infof("Created request to GET %s", req.Raw().URL.String())
+	return req, nil
+}
+
+func (ri *ARMResourceImporter) Client() *azruntime.Pipeline {
+	return ri.client
+}
+
+func (ri *ARMResourceImporter) Config() cloud.ServiceConfiguration {
+	return ri.serviceConfiguration
+}
+
+func (ri *ARMResourceImporter) createBlankObjectFromID(armID *arm.ResourceID) (runtime.Object, error) {
+	gvk, err := ri.groupVersionKindFromID(armID)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to determine GVK of resource")
+	}
+
+	obj, err := ri.createBlankObjectFromGVK(gvk)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create blank resource")
+	}
+
+	if mo, ok := obj.(genruntime.ARMMetaObject); ok {
+		name := ri.nameFromID(armID)
+		mo.SetName(name)
+	}
+
+	return obj, nil
+}
+
+// groupVersionKindFromID returns the GroupVersionKind for the resource we're importing
+func (ri *ARMResourceImporter) groupVersionKindFromID(id *arm.ResourceID) (schema.GroupVersionKind, error) {
+	gk, err := ri.groupKindFromID(id)
+	if err != nil {
+		return schema.GroupVersionKind{}, err
+	}
+
+	return ri.selectVersionFromGK(gk)
+}
+
+// groupKindFromID parses a GroupKind from the resource URL, allowing us to look up the actual resource
+func (ri *ARMResourceImporter) groupKindFromID(id *arm.ResourceID) (schema.GroupKind, error) {
+	t := id.ResourceType.String()
+	if t == "" {
+		return schema.GroupKind{}, errors.Errorf("unable to determine resource type from ID %s", id)
+	}
+
+	gk, ok := typesToGK[t]
+	if !ok {
+		return schema.GroupKind{}, errors.Errorf("unable to determine resource type from ID %s", id)
+	}
+
+	return gk, nil
+}
+
+func (ri *ARMResourceImporter) nameFromID(id *arm.ResourceID) string {
+	klog.V(3).Infof("Name: %s", id.Name)
+	return id.Name
+}
+
+func CreateARMClient(cloudConfig cloud.Configuration) (*azruntime.Pipeline, error) {
+	creds, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get default azure credential")
+	}
+
+	var userAgent = "asoctl/" + version.BuildVersion
+
+	opts := &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Cloud: cloudConfig,
+			PerCallPolicies: []policy.Policy{
+				genericarmclient.NewUserAgentPolicy(userAgent),
+			},
+		},
+	}
+
+	pipeline, err := armruntime.NewPipeline("generic", version.BuildVersion, creds, azruntime.PipelineOptions{}, opts)
+	return &pipeline, err
+}
+
+var typesToGK map[string]schema.GroupKind
+
+func init() {
```

Comment:
Whats the reason for using init here? Not sure if it'll always run even if the user is not using the resource import command?

Reply:
Agree with @super-harsh, we should steer clear of init.

We could do a lazily initialized variable with something like `sync.Once` maybe?

Reply:
Now using lazy initialization.